backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/StaticFileMapper.java

```diff
@@ -71,6 +79,34 @@ private static void redirectAssets(Context ctx) {
   }
 
   private static void returnIndexFile(Context ctx) {
+    // redirect apps from /schema/app to /schema/app
+    if (!ctx.path().endsWith("/")) {
+      try {
+        String unescapedPath = URLDecoder.decode(ctx.path(), StandardCharsets.UTF_8.name());
+        // validate valid app path
+        List<String> path =
+            Arrays.stream(unescapedPath.split("/")).filter(item -> item.length() > 0).toList();
+        if (path.size() != 2) {
+          throw new MolgenisException("invalid app path. Needs to be /schema/app");
+        }
+        // validate that it is a valid schema name
+        if (!path.get(0).matches(Constants.SCHEMA_NAME_REGEX)) {
+          throw new MolgenisException(
+              "path invalid: schema parameter is invalid. Please provide a valid /schema/app");
+        }
+        // validate that it is a known app
+        URL app = StaticFileMapper.class.getResource("/public_html/apps/" + path.get(1));
```

Comment:
This might open the possibility to probe the filesystem.

Reply:
how? it only checks if apps exist nothing more. See tests.